modules/oas-page-builder/src/services/pageBuilder.ts

```diff
@@ -0,0 +1,82 @@
+import fetch from 'node-fetch';
+import { ModuleOptions } from '../types';
+import { normalizePath } from '../utils/normalizePath';
+import { execRedoc } from './commandExecutor';
+import { findLastSavedGitHash } from './database';
+import { OASPageMetadata } from './types';
+
+const fetchTextData = async (url: string, errMsg: string) => {
+  const res = await fetch(url);
+  if (!res.ok) {
+    // Error should be caught when creating pages.
+    throw new Error(`${errMsg}; ${res.statusText}`);
+  }
+  return res.text();
+};
+
+const getOASFileUrl = (gitHash: string) =>
+  `https://mongodb-mms-prod-build-server.s3.amazonaws.com/openapi/${gitHash}.json`;
```

Comment:
We might want to consider parameterizing, or at least storing the url for this string this in a constant e.g. `const OAS_FILE_SERVER = https://mongodb-mms-prod-build-server.s3.amazonaws.com/openapi/`

Reply:
Done. Leaving as a constant for now. We should be able to easily parameterize it in the future depending on what URLs we need